src/main/java/org/helpberkeley/memberdata/WorkflowParser.java

```diff
@@ -51,23 +51,18 @@ protected WorkflowParser(final String csvData) {
     public static WorkflowParser create(Map<String, Restaurant> globalRestaurants, String csvData) {
 
         ControlBlock controlBlock = ControlBlock.create(csvData);
+        String version = controlBlock.getVersion();
         WorkflowParser workflowParser;
 
-        switch (controlBlock.getVersion()) {
-            case Constants.CONTROL_BLOCK_VERSION_UNKNOWN:
-                throw new MemberDataException("Control block not found");
-            case Constants.CONTROL_BLOCK_VERSION_200:
-                workflowParser = new WorkflowParserV200(csvData);
-                break;
-            case Constants.CONTROL_BLOCK_VERSION_300:
-            case Constants.CONTROL_BLOCK_VERSION_301:
-            case Constants.CONTROL_BLOCK_VERSION_302:
-                workflowParser = new WorkflowParserV300(csvData);
-                break;
-            case Constants.CONTROL_BLOCK_VERSION_1:
-            default:
-                throw new MemberDataException(
-                        "Control block version " + controlBlock.getVersion() + " is not supported.\n");
+        if (version.equals(Constants.CONTROL_BLOCK_VERSION_UNKNOWN)) {
+            throw new MemberDataException("Control block not found");
+        } else if (controlBlock.versionIsCompatible(Constants.CONTROL_BLOCK_VERSION_200)) {
+            workflowParser = new WorkflowParserV200(csvData);
+        } else if (controlBlock.versionIsCompatible(Constants.CONTROL_BLOCK_VERSION_300)) {
+            workflowParser = new WorkflowParserV300(csvData);
```

Comment:
Since we have actual 202, 301, and 302 ControlBlock instances, I wonder if it would be better here to create the actual version ControlBlock, and use versionIsCompatible in other places that need to steer the logic to 200 vs. 300.   Thoughts?

Reply:
I'm not sure I understand what you mean here. The control block is being created just above - it could be v202/301/etc - but we only have 2 versions of WorkflowParser so we need to steer it to one of those, right?